lib/AST/TypeCheckRequests.cpp

```diff
@@ -2075,3 +2075,62 @@ void SemanticDeclAttrsRequest::cacheResult(DeclAttributes attrs) const {
   auto decl = std::get<0>(getStorage());
   const_cast<Decl *>(decl)->setSemanticAttrsComputed(true);
 }
+
+//----------------------------------------------------------------------------//
+// UniqueUnderlyingTypeSubstitutionsRequest computation.
+//----------------------------------------------------------------------------//
+
+static bool
+shouldTypecheckFunctionBodyForUniqueUnderlyingType(AbstractFunctionDecl *afd) {
+  auto mod = afd->getModuleContext();
+  if (!mod->isMainModule())
+    return true;
+
+  // If the main module has no primary source files then the compilation is a
+  // whole module build and all source files can be typechecked.
+  if (mod->getPrimarySourceFiles().size() == 0)
+    return true;
+
+  auto sf = afd->getParentSourceFile();
+  if (!sf)
+    return true;
+
+  if (sf->getKind() != FileUnitKind::Source)
```

Comment:
I believe `isa<SourceFile>` if and only if `getKind() == FileUnitKind::Source`

Reply:
Good catch, I meant to be testing the SourceFileKind here, not the FileUnitKind